packages/react-chat/src/contexts/RuntimeContext/useRuntimeState.ts

```diff
@@ -35,15 +37,17 @@ export const useRuntimeState = ({ assistant, config }: Settings) => {
 
   const [indicator, setIndicator] = useState(false);
 
-  // TODO: refactor no-reply system
-  const [lastInteractionAt, setLastInteractionAt] = useState<number | null>(Date.now());
-  const [noReplyTimeout, setNoReplyTimeout] = useState<number | null>(null);
+  const { clearNoReplyTimeout, setNoReplyTimeout } = useNoReply(() => ({ interact, isStatus }));
+
   const noReplyHandler: TraceDeclaration<RuntimeMessage, any> = {
     canHandle: ({ type }) => type === ActionType.NO_REPLY,
     handle: ({ context }, trace: Trace.NoReplyTrace) => {
       if (trace.payload?.timeout) {
-        setNoReplyTimeout(trace.payload.timeout * 1000);
-        setLastInteractionAt(Date.now());
+        // messages take 1 second to animate in, on top of the delay
+        const messageDelays = context.messages.reduce((acc, message) => acc + (message.delay ?? 1000) + DEFAULT_MESSAGE_DELAY, 0);
```

Comment:
on top of the actual delay, there is a second animation delay. I did some tests to make sure this was the case:
https://github.com/voiceflow/react-chat/blob/7ae16de36b0fd0fd48e059f06f78c6dbf8e5e636/packages/react-chat/src/components/SystemResponse/hooks.ts#L63-L73

You can see that the default delay to animate in something is actually 2 seconds, not 1

Reply:
`context.messages` only contains messages that will actually be displayed. The runtime also guarantees that `NoReply` is the last trace.